### PR TITLE
[Compiler-v2] Fix native struct

### LINKED ISSUE #13758: [Bug][Compiler-v2] ZERO_SIZED_STRUCT raised in `multi_pool_money_market_token`
# 🐛 Bug

When compiling the test `multi_pool_money_market_token` using V2, the bytecode verification error `ZERO_SIZED_STRUCT` is raised.

### PR DESCRIPTION
**Pull request by rahxephon89**
## Description

This PR closes #13758 by setting field_information as Native instead of Declared when the field count is 0 (which means it is a native struct).

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

1) existing tests pass;
2) migrate two tests from V1.

## Key Areas to Review

Since `dummy_field` is inserted for user defined empty structs during compilation, if the field count is 0, the struct is native. Need to guarantee that this reasoning is correct.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs [hunk 1]
```diff
@@ -217,20 +217,24 @@ impl ModuleGenerator {
...
-                        .expect(SOURCE_MAP_OK);
-                    let name = self.name_index(ctx, field_loc, f.get_name());
-                    let signature =
-                        FF::TypeSignature(self.signature_token(ctx, loc, &f.get_type()));
-                    FF::FieldDefinition { name, signature }
-                })
-                .collect(),
-        );
+        let field_information = if struct_env.get_field_count() == 0 {
+            FF::StructFieldInformation::Native
```
Comment (brmataptos) [inline third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs:221]: Can you explain why Native is the right choice here?  How is it treated differently and where?

We also might want a comment here so this is not undone in the future by someone who doesn't understand why you did it.

Comment (rahxephon89) [reply third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs:221]: If a user-defined struct has no field, a `dummy_field` will be added. So if a struct does not have any field, it means it is a native struct. Also, the bytecode verifier will generate the error `ZERO_SIZED_STRUCT` for a struct with no fields if it is not native. I will add a comment here.

Comment (vineethk) [reply third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs:223]: We should ideally not resort to secondary "happenstance" information (i.e., invariants that hold today, but may not under future refactorings), especially for information that is explicitly specified (all `native` structs have to be explicitly mentioned as such in the code, right)?

For functions, we maintain information about whether is is native. For structs, we should analogously also maintain such information (I am assuming it is not in the move model, which may have just been an oversight). Then, we can use that explicit information here.

**Review by brmataptos — APPROVED** [review]

File: third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs (hunk 1 above)
Comment (rahxephon89) [reply third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs:221]: Thanks @vineethk. I added the field `is_native` to explicitly mark a native struct. PTAL

**Review by wrwg — APPROVED** [review]

**Review by vineethk — APPROVED** [review]

**Pull request merged**